cupy/_core/_routines_manipulation.pyx

```diff
@@ -57,13 +57,7 @@ cdef _ndarray_shape_setter(ndarray self, newshape):
         newshape = (newshape,)
     shape = internal.infer_unknown_dimension(newshape, self.size)
     _get_strides_for_nocopy_reshape(self, shape, strides)
-    if strides.size() != shape.size():
-        raise AttributeError(
-            'Incompatible shape for in-place modification. Use `.reshape()` '
-            'to make a copy with the desired shape.')
```

Comment:
Hi @toslunar, I noticed you touched this error message in #5470, but it seems to me there is no way to trigger this error. Could you confirm?

Reply:
I should have added a test.

```
>>> x = cupy.zeros((2, 3), order='F')
>>> x.shape = (3, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "cupy/_core/core.pyx", line 297, in cupy._core.core.ndarray.shape.__set__
  File "cupy/_core/_routines_manipulation.pyx", line 61, in cupy._core._routines_manipulation._ndarray_shape_setter
AttributeError: Incompatible shape for in-place modification. Use `.reshape()` to make a copy with the desired shape.
```

Reply:
Thanks, @toslunar. No worries, let me add it.

Reply:
Done in a0e1ae5.